### PR TITLE
New API integration

### PR DESCRIPTION
**Pull request by developerfromjokela**
I've integrated new API into use for getting prices, fix for #412 

New API doesn't have region names like Oslo, etc for norway. They're NO1, NO2, etc instead.

### CODE REVIEW

**Comment by developerfromjokela** [comment]
Näyttökuva 2024-10-08 kello 18 37 35

Tested on own HA instance, seems to work. 
More testing would be appreciated

File: custom_components/nordpool/test_parser.py [hunk 1]
```diff
@@ -1,42 +1,230 @@
...
+    async def _io(self, url, **kwargs):
+
+        resp = await self.client.get(url, params=kwargs, headers={
+            'Origin': 'https://data.nordpoolgroup.com'
+        })
+        _LOGGER.debug("requested %s %s", resp.url, kwargs)
+
+        return await resp.json()
+
+    def _parse_json(self, data, areas=[]):
```
Comment (Hellowlol) [inline custom_components/nordpool/test_parser.py:48]: Change area to None and the check if areas is None. We do not want mutable keyword arguments

Comment (developerfromjokela) [reply custom_components/nordpool/test_parser.py:48]: What is test_parser used for?

**Comment by Hellowlol** [comment]
Thanks for taking time to create a PR for this. Sorry for the lack of description on the review. Im on mobile

**Review by simonepittis — APPROVED** [review]
Test Ok form me

**Comment by developerfromjokela** [comment]
There seems to be a single entry for tomorrow, even though no prices are available for tomorrow yet.

**Comment by developerfromjokela** [comment]
> we can see the price for tomorrow only after 13:30 CET

Yup, but here's how mine looks atm:

image

**Comment by developerfromjokela** [comment]
Technically it's raw data, there's no entry in the actual tomorrow attribute. I just have some graphs that use raw data :)

**Comment by dana-se** [comment]
Not here, with 0.0.15b2 version

image

**Comment by simonepittis** [comment]
tested with plugin "Energi Data Service", same result:
image
the data don't comes from NordPool

**Comment by developerfromjokela** [comment]
Näyttökuva 2024-10-09 kello 11 43 33

https://data.nordpoolgroup.com/auction/day-ahead/prices?deliveryDate=2024-10-09&currency=EUR&aggregation=Hourly&deliveryAreas=FI it's also present on their site with same API. Maybe just should ignore last value, so no items in raw_tomorrow?

**Comment by dana-se** [comment]
image

https://data.nordpoolgroup.com/auction/day-ahead/prices?deliveryDate=2024-10-09&currency=EUR&aggregation=Hourly&deliveryAreas=FI

This is how it looks for me.

**Comment by developerfromjokela** [comment]
> image
> 
> https://data.nordpoolgroup.com/auction/day-ahead/prices?deliveryDate=2024-10-09&currency=EUR&aggregation=Hourly&deliveryAreas=FI
> 
> This is how it looks for me.

Strange. Only difference I see between you and me is timezone. Is this a bug in nordpool? 😆

**Comment by dana-se** [comment]
Is it the same issue, if you open this link?
https://data.nordpoolgroup.com/auction/day-ahead/prices?deliveryDate=latest&currency=SEK&aggregation=Hourly&deliveryAreas=SE4

**Comment by developerfromjokela** [comment]
Could it be that the data is not in UTC, but CET?

**Comment by developerfromjokela** [comment]
I played with timezone options and with CET I get same values as you.
Näyttökuva 2024-10-09 kello 12 26 57

**Comment by dana-se** [comment]
> I played with timezone options and with CET I get same values as you. Näyttökuva 2024-10-09 kello 12 26 57

Yes, that's the same settings that I have.

**Comment by developerfromjokela** [comment]
It seems like nordpool's data ranges from 01:00 - 01:00 not from 00:00-00:00.

**Comment by developerfromjokela** [comment]
Ok I think I figured out what happened. Dateutils parsed timestamps as UTC from API, even though they're Europe/Stockholm. I added an override for parsing dates and now at least FI prices align with what other sources are reporting.

**Comment by developerfromjokela** [comment]
I'm trying to debug another issue, after reloading the integration, my raw_today fills with over 100 entries of today's prices and many duplicates of tomorrow's prices.

**Comment by developerfromjokela** [comment]
Here's what happens:

```yaml
today: 0, 0.001, 0, 0, 0, 0.001, 0.423, 0.611, 1.236, 1.403, 1.431, 1.408, 1.177, 0.999, 0.906, 0.949, 1.015, 1.118, 1.186, 1.093, 0.818, 0.19, 0.001, 0
tomorrow: -0.001, -0.073, -0.102, -0.103, -0.1, -0.039, 0, 1.268, 2.845, 3.028, 3.141, 3.135, 3.105, 2.74, 2, 2.482, 1.884, 1.751, 2.007, 1.851, 0.25, 0, -0.001, -0.013
tomorrow_valid: true
raw_today: 
- start: '2024-10-09T00:00:00+03:00'
  end: '2024-10-09T01:00:00+03:00'
  value: 0
- start: '2024-10-09T01:00:00+03:00'
  end: '2024-10-09T02:00:00+03:00'
  value: 0.001
- start: '2024-10-09T02:00:00+03:00'
  end: '2024-10-09T03:00:00+03:00'
  value: 0
- start: '2024-10-09T03:00:00+03:00'
  end: '2024-10-09T04:00:00+03:00'
  value: 0
- start: '2024-10-09T04:00:00+03:00'
  end: '2024-10-09T05:00:00+03:00'
  value: 0
- start: '2024-10-09T05:00:00+03:00'
  end: '2024-10-09T06:00:00+03:00'
  value: 0.001
- start: '2024-10-09T06:00:00+03:00'
  end: '2024-10-09T07:00:00+03:00'
  value: 0.423
- start: '2024-10-09T07:00:00+03:00'
  end: '2024-10-09T08:00:00+03:00'
  value: 0.611
- start: '2024-10-09T08:00:00+03:00'
  end: '2024-10-09T09:00:00+03:00'
  value: 1.236
- start: '2024-10-09T09:00:00+03:00'
  end: '2024-10-09T10:00:00+03:00'
  value: 1.403
- start: '2024-10-09T10:00:00+03:00'
  end: '2024-10-09T11:00:00+03:00'
  value: 1.431
- start: '2024-10-09T11:00:00+03:00'
  end: '2024-10-09T12:00:00+03:00'
  value: 1.408
- start: '2024-10-09T12:00:00+03:00'
  end: '2024-10-09T13:00:00+03:00'
  value: 1.177
- start: '2024-10-09T13:00:00+03:00'
  end: '2024-10-09T14:00:00+03:00'
  value: 0.999
- start: '2024-10-09T14:00:00+03:00'
  end: '2024-10-09T15:00:00+03:00'
  value: 0.906
- start: '2024-10-09T15:00:00+03:00'
  end: '2024-10-09T16:00:00+03:00'
  value: 0.949
- start: '2024-10-09T16:00:00+03:00'
  end: '2024-10-09T17:00:00+03:00'
  value: 1.015
- start: '2024-10-09T17:00:00+03:00'
  end: '2024-10-09T18:00:00+03:00'
  value: 1.118
- start: '2024-10-09T18:00:00+03:00'
  end: '2024-10-09T19:00:00+03:00'
  value: 1.186
- start: '2024-10-09T19:00:00+03:00'
  end: '2024-10-09T20:00:00+03:00'
  value: 1.093
- start: '2024-10-09T20:00:00+03:00'
  end: '2024-10-09T21:00:00+03:00'
  value: 0.818
- start: '2024-10-09T21:00:00+03:00'
  end: '2024-10-09T22:00:00+03:00'
  value: 0.19
- start: '2024-10-09T22:00:00+03:00'
  end: '2024-10-09T23:00:00+03:00'
  value: 0.001
- start: '2024-10-09T23:00:00+03:00'
  end: '2024-10-10T00:00:00+03:00'
  value: 0
- start: '2024-10-10T00:00:00+03:00'
  end: '2024-10-10T01:00:00+03:00'
  value: -0.001
- start: '2024-10-10T01:00:00+03:00'
  end: '2024-10-10T02:00:00+03:00'
  value: -0.073
- start: '2024-10-10T02:00:00+03:00'
  end: '2024-10-10T03:00:00+03:00'
  value: -0.102
- start: '2024-10-10T03:00:00+03:00'
  end: '2024-10-10T04:00:00+03:00'
  value: -0.103
- start: '2024-10-10T04:00:00+03:00'
  end: '2024-10-10T05:00:00+03:00'
  value: -0.1
- start: '2024-10-10T05:00:00+03:00'
  end: '2024-10-10T06:00:00+03:00'
  value: -0.039
- start: '2024-10-10T06:00:00+03:00'
  end: '2024-10-10T07:00:00+03:00'
  value: 0
- start: '2024-10-10T07:00:00+03:00'
  end: '2024-10-10T08:00:00+03:00'
  value: 1.268
- start: '2024-10-10T08:00:00+03:00'
  end: '2024-10-10T09:00:00+03:00'
  value: 2.845
- start: '2024-10-10T09:00:00+03:00'
  end: '2024-10-10T10:00:00+03:00'
  value: 3.028
- start: '2024-10-10T10:00:00+03:00'
  end: '2024-10-10T11:00:00+03:00'
  value: 3.141
- start: '2024-10-10T11:00:00+03:00'
  end: '2024-10-10T12:00:00+03:00'
  value: 3.135
- start: '2024-10-10T12:00:00+03:00'
  end: '2024-10-10T13:00:00+03:00'
  value: 3.105
- start: '2024-10-10T13:00:00+03:00'
  end: '2024-10-10T14:00:00+03:00'
  value: 2.74
- start: '2024-10-10T14:00:00+03:00'
  end: '2024-10-10T15:00:00+03:00'
  value: 2
- start: '2024-10-10T15:00:00+03:00'
  end: '2024-10-10T16:00:00+03:00'
  value: 2.482
- start: '2024-10-10T16:00:00+03:00'
  end: '2024-10-10T17:00:00+03:00'
  value: 1.884
- start: '2024-10-10T17:00:00+03:00'
  end: '2024-10-10T18:00:00+03:00'
  value: 1.751
- start: '2024-10-10T18:00:00+03:00'
  end: '2024-10-10T19:00:00+03:00'
  value: 2.007
- start: '2024-10-10T19:00:00+03:00'
  end: '2024-10-10T20:00:00+03:00'
  value: 1.851
- start: '2024-10-10T20:00:00+03:00'
  end: '2024-10-10T21:00:00+03:00'
  value: 0.25
- start: '2024-10-10T21:00:00+03:00'
  end: '2024-10-10T22:00:00+03:00'
  value: 0
- start: '2024-10-10T22:00:00+03:00'
  end: '2024-10-10T23:00:00+03:00'
  value: -0.001
- start: '2024-10-10T23:00:00+03:00'
  end: '2024-10-11T00:00:00+03:00'
  value: -0.013
- start: '2024-10-10T00:00:00+03:00'
  end: '2024-10-10T01:00:00+03:00'
  value: -0.001
- start: '2024-10-10T01:00:00+03:00'
  end: '2024-10-10T02:00:00+03:00'
  value: -0.073
- start: '2024-10-10T02:00:00+03:00'
  end: '2024-10-10T03:00:00+03:00'
  value: -0.102
- start: '2024-10-10T03:00:00+03:00'
  end: '2024-10-10T04:00:00+03:00'
  value: -0.103
- start: '2024-10-10T04:00:00+03:00'
  end: '2024-10-10T05:00:00+03:00'
  value: -0.1
- start: '2024-10-10T05:00:00+03:00'
  end: '2024-10-10T06:00:00+03:00'
  value: -0.039
- start: '2024-10-10T06:00:00+03:00'
  end: '2024-10-10T07:00:00+03:00'
  value: 0
- start: '2024-10-10T07:00:00+03:00'
  end: '2024-10-10T08:00:00+03:00'
  value: 1.268
- start: '2024-10-10T08:00:00+03:00'
  end: '2024-10-10T09:00:00+03:00'
  value: 2.845
- start: '2024-10-10T09:00:00+03:00'
  end: '2024-10-10T10:00:00+03:00'
  value: 3.028
- start: '2024-10-10T10:00:00+03:00'
  end: '2024-10-10T11:00:00+03:00'
  value: 3.141
- start: '2024-10-10T11:00:00+03:00'
  end: '2024-10-10T12:00:00+03:00'
  value: 3.135
- start: '2024-10-10T12:00:00+03:00'
  end: '2024-10-10T13:00:00+03:00'
  value: 3.105
- start: '2024-10-10T13:00:00+03:00'
  end: '2024-10-10T14:00:00+03:00'
  value: 2.74
- start: '2024-10-10T14:00:00+03:00'
  end: '2024-10-10T15:00:00+03:00'
  value: 2
- start: '2024-10-10T15:00:00+03:00'
  end: '2024-10-10T16:00:00+03:00'
  value: 2.482
- start: '2024-10-10T16:00:00+03:00'
  end: '2024-10-10T17:00:00+03:00'
  value: 1.884
- start: '2024-10-10T17:00:00+03:00'
  end: '2024-10-10T18:00:00+03:00'
  value: 1.751
- start: '2024-10-10T18:00:00+03:00'
  end: '2024-10-10T19:00:00+03:00'
  value: 2.007
- start: '2024-10-10T19:00:00+03:00'
  end: '2024-10-10T20:00:00+03:00'
  value: 1.851
- start: '2024-10-10T20:00:00+03:00'
  end: '2024-10-10T21:00:00+03:00'
  value: 0.25
- start: '2024-10-10T21:00:00+03:00'
  end: '2024-10-10T22:00:00+03:00'
  value: 0
- start: '2024-10-10T22:00:00+03:00'
  end: '2024-10-10T23:00:00+03:00'
  value: -0.001
- start: '2024-10-10T23:00:00+03:00'
  end: '2024-10-11T00:00:00+03:00'
  value: -0.013
- start: '2024-10-10T00:00:00+03:00'
  end: '2024-10-10T01:00:00+03:00'
  value: -0.001
- start: '2024-10-10T01:00:00+03:00'
  end: '2024-10-10T02:00:00+03:00'
  value: -0.073
- start: '2024-10-10T02:00:00+03:00'
  end: '2024-10-10T03:00:00+03:00'
  value: -0.102
- start: '2024-10-10T03:00:00+03:00'
  end: '2024-10-10T04:00:00+03:00'
  value: -0.103
- start: '2024-10-10T04:00:00+03:00'
  end: '2024-10-10T05:00:00+03:00'
  value: -0.1
- start: '2024-10-10T05:00:00+03:00'
  end: '2024-10-10T06:00:00+03:00'
  value: -0.039
- start: '2024-10-10T06:00:00+03:00'
  end: '2024-10-10T07:00:00+03:00'
  value: 0
- start: '2024-10-10T07:00:00+03:00'
  end: '2024-10-10T08:00:00+03:00'
  value: 1.268
- start: '2024-10-10T08:00:00+03:00'
  end: '2024-10-10T09:00:00+03:00'
  value: 2.845
- start: '2024-10-10T09:00:00+03:00'
  end: '2024-10-10T10:00:00+03:00'
  value: 3.028
- start: '2024-10-10T10:00:00+03:00'
  end: '2024-10-10T11:00:00+03:00'
  value: 3.141
- start: '2024-10-10T11:00:00+03:00'
  end: '2024-10-10T12:00:00+03:00'
  value: 3.135
- start: '2024-10-10T12:00:00+03:00'
  end: '2024-10-10T13:00:00+03:00'
  value: 3.105
- start: '2024-10-10T13:00:00+03:00'
  end: '2024-10-10T14:00:00+03:00'
  value: 2.74
- start: '2024-10-10T14:00:00+03:00'
  end: '2024-10-10T15:00:00+03:00'
  value: 2
- start: '2024-10-10T15:00:00+03:00'
  end: '2024-10-10T16:00:00+03:00'
  value: 2.482
- start: '2024-10-10T16:00:00+03:00'
  end: '2024-10-10T17:00:00+03:00'
  value: 1.884
- start: '2024-10-10T17:00:00+03:00'
  end: '2024-10-10T18:00:00+03:00'
  value: 1.751
- start: '2024-10-10T18:00:00+03:00'
  end: '2024-10-10T19:00:00+03:00'
  value: 2.007
- start: '2024-10-10T19:00:00+03:00'
  end: '2024-10-10T20:00:00+03:00'
  value: 1.851
- start: '2024-10-10T20:00:00+03:00'
  end: '2024-10-10T21:00:00+03:00'
  value: 0.25
- start: '2024-10-10T21:00:00+03:00'
  end: '2024-10-10T22:00:00+03:00'
  value: 0
- start: '2024-10-10T22:00:00+03:00'
  end: '2024-10-10T23:00:00+03:00'
  value: -0.001
- start: '2024-10-10T23:00:00+03:00'
  end: '2024-10-11T00:00:00+03:00'
  value: -0.013
- start: '2024-10-10T00:00:00+03:00'
  end: '2024-10-10T01:00:00+03:00'
  value: -0.001
- start: '2024-10-10T01:00:00+03:00'
  end: '2024-10-10T02:00:00+03:00'
  value: -0.073
- start: '2024-10-10T02:00:00+03:00'
  end: '2024-10-10T03:00:00+03:00'
  value: -0.102
- start: '2024-10-10T03:00:00+03:00'
  end: '2024-10-10T04:00:00+03:00'
  value: -0.103
- start: '2024-10-10T04:00:00+03:00'
  end: '2024-10-10T05:00:00+03:00'
  value: -0.1
- start: '2024-10-10T05:00:00+03:00'
  end: '2024-10-10T06:00:00+03:00'
  value: -0.039
- start: '2024-10-10T06:00:00+03:00'
  end: '2024-10-10T07:00:00+03:00'
  value: 0
- start: '2024-10-10T07:00:00+03:00'
  end: '2024-10-10T08:00:00+03:00'
  value: 1.268
- start: '2024-10-10T08:00:00+03:00'
  end: '2024-10-10T09:00:00+03:00'
  value: 2.845
- start: '2024-10-10T09:00:00+03:00'
  end: '2024-10-10T10:00:00+03:00'
  value: 3.028
- start: '2024-10-10T10:00:00+03:00'
  end: '2024-10-10T11:00:00+03:00'
  value: 3.141
- start: '2024-10-10T11:00:00+03:00'
  end: '2024-10-10T12:00:00+03:00'
  value: 3.135
- start: '2024-10-10T12:00:00+03:00'
  end: '2024-10-10T13:00:00+03:00'
  value: 3.105
- start: '2024-10-10T13:00:00+03:00'
  end: '2024-10-10T14:00:00+03:00'
  value: 2.74
- start: '2024-10-10T14:00:00+03:00'
  end: '2024-10-10T15:00:00+03:00'
  value: 2
- start: '2024-10-10T15:00:00+03:00'
  end: '2024-10-10T16:00:00+03:00'
  value: 2.482
- start: '2024-10-10T16:00:00+03:00'
  end: '2024-10-10T17:00:00+03:00'
  value: 1.884
- start: '2024-10-10T17:00:00+03:00'
  end: '2024-10-10T18:00:00+03:00'
  value: 1.751
- start: '2024-10-10T18:00:00+03:00'
  end: '2024-10-10T19:00:00+03:00'
  value: 2.007
- start: '2024-10-10T19:00:00+03:00'
  end: '2024-10-10T20:00:00+03:00'
  value: 1.851
- start: '2024-10-10T20:00:00+03:00'
  end: '2024-10-10T21:00:00+03:00'
  value: 0.25
- start: '2024-10-10T21:00:00+03:00'
  end: '2024-10-10T22:00:00+03:00'
  value: 0
- start: '2024-10-10T22:00:00+03:00'
  end: '2024-10-10T23:00:00+03:00'
  value: -0.001
- start: '2024-10-10T23:00:00+03:00'
  end: '2024-10-11T00:00:00+03:00'
  value: -0.013
- start: '2024-10-10T00:00:00+03:00'
  end: '2024-10-10T01:00:00+03:00'
  value: -0.001
- start: '2024-10-10T01:00:00+03:00'
  end: '2024-10-10T02:00:00+03:00'
  value: -0.073
- start: '2024-10-10T02:00:00+03:00'
  end: '2024-10-10T03:00:00+03:00'
  value: -0.102
- start: '2024-10-10T03:00:00+03:00'
  end: '2024-10-10T04:00:00+03:00'
  value: -0.103
- start: '2024-10-10T04:00:00+03:00'
  end: '2024-10-10T05:00:00+03:00'
  value: -0.1
- start: '2024-10-10T05:00:00+03:00'
  end: '2024-10-10T06:00:00+03:00'
  value: -0.039
- start: '2024-10-10T06:00:00+03:00'
  end: '2024-10-10T07:00:00+03:00'
  value: 0
- start: '2024-10-10T07:00:00+03:00'
  end: '2024-10-10T08:00:00+03:00'
  value: 1.268
- start: '2024-10-10T08:00:00+03:00'
  end: '2024-10-10T09:00:00+03:00'
  value: 2.845
- start: '2024-10-10T09:00:00+03:00'
  end: '2024-10-10T10:00:00+03:00'
  value: 3.028
- start: '2024-10-10T10:00:00+03:00'
  end: '2024-10-10T11:00:00+03:00'
  value: 3.141
- start: '2024-10-10T11:00:00+03:00'
  end: '2024-10-10T12:00:00+03:00'
  value: 3.135
- start: '2024-10-10T12:00:00+03:00'
  end: '2024-10-10T13:00:00+03:00'
  value: 3.105
- start: '2024-10-10T13:00:00+03:00'
  end: '2024-10-10T14:00:00+03:00'
  value: 2.74
- start: '2024-10-10T14:00:00+03:00'
  end: '2024-10-10T15:00:00+03:00'
  value: 2
- start: '2024-10-10T15:00:00+03:00'
  end: '2024-10-10T16:00:00+03:00'
  value: 2.482
- start: '2024-10-10T16:00:00+03:00'
  end: '2024-10-10T17:00:00+03:00'
  value: 1.884
- start: '2024-10-10T17:00:00+03:00'
  end: '2024-10-10T18:00:00+03:00'
  value: 1.751
- start: '2024-10-10T18:00:00+03:00'
  end: '2024-10-10T19:00:00+03:00'
  value: 2.007
- start: '2024-10-10T19:00:00+03:00'
  end: '2024-10-10T20:00:00+03:00'
  value: 1.851
- start: '2024-10-10T20:00:00+03:00'
  end: '2024-10-10T21:00:00+03:00'
  value: 0.25
- start: '2024-10-10T21:00:00+03:00'
  end: '2024-10-10T22:00:00+03:00'
  value: 0
- start: '2024-10-10T22:00:00+03:00'
  end: '2024-10-10T23:00:00+03:00'
  value: -0.001
- start: '2024-10-10T23:00:00+03:00'
  end: '2024-10-11T00:00:00+03:00'
  value: -0.013
- start: '2024-10-10T00:00:00+03:00'
  end: '2024-10-10T01:00:00+03:00'
  value: -0.001
- start: '2024-10-10T01:00:00+03:00'
  end: '2024-10-10T02:00:00+03:00'
  value: -0.073
- start: '2024-10-10T02:00:00+03:00'
  end: '2024-10-10T03:00:00+03:00'
  value: -0.102
- start: '2024-10-10T03:00:00+03:00'
  end: '2024-10-10T04:00:00+03:00'
  value: -0.103
- start: '2024-10-10T04:00:00+03:00'
  end: '2024-10-10T05:00:00+03:00'
  value: -0.1
- start: '2024-10-10T05:00:00+03:00'
  end: '2024-10-10T06:00:00+03:00'
  value: -0.039
- start: '2024-10-10T06:00:00+03:00'
  end: '2024-10-10T07:00:00+03:00'
  value: 0
- start: '2024-10-10T07:00:00+03:00'
  end: '2024-10-10T08:00:00+03:00'
  value: 1.268
- start: '2024-10-10T08:00:00+03:00'
  end: '2024-10-10T09:00:00+03:00'
  value: 2.845
- start: '2024-10-10T09:00:00+03:00'
  end: '2024-10-10T10:00:00+03:00'
  value: 3.028
- start: '2024-10-10T10:00:00+03:00'
  end: '2024-10-10T11:00:00+03:00'
  value: 3.141
- start: '2024-10-10T11:00:00+03:00'
  end: '2024-10-10T12:00:00+03:00'
  value: 3.135
- start: '2024-10-10T12:00:00+03:00'
  end: '2024-10-10T13:00:00+03:00'
  value: 3.105
- start: '2024-10-10T13:00:00+03:00'
  end: '2024-10-10T14:00:00+03:00'
  value: 2.74
- start: '2024-10-10T14:00:00+03:00'
  end: '2024-10-10T15:00:00+03:00'
  value: 2
- start: '2024-10-10T15:00:00+03:00'
  end: '2024-10-10T16:00:00+03:00'
  value: 2.482
- start: '2024-10-10T16:00:00+03:00'
  end: '2024-10-10T17:00:00+03:00'
  value: 1.884
- start: '2024-10-10T17:00:00+03:00'
  end: '2024-10-10T18:00:00+03:00'
  value: 1.751
- start: '2024-10-10T18:00:00+03:00'
  end: '2024-10-10T19:00:00+03:00'
  value: 2.007
- start: '2024-10-10T19:00:00+03:00'
  end: '2024-10-10T20:00:00+03:00'
  value: 1.851
- start: '2024-10-10T20:00:00+03:00'
  end: '2024-10-10T21:00:00+03:00'
  value: 0.25
- start: '2024-10-10T21:00:00+03:00'
  end: '2024-10-10T22:00:00+03:00'
  value: 0
- start: '2024-10-10T22:00:00+03:00'
  end: '2024-10-10T23:00:00+03:00'
  value: -0.001
- start: '2024-10-10T23:00:00+03:00'
  end: '2024-10-11T00:00:00+03:00'
  value: -0.013
- start: '2024-10-10T00:00:00+03:00'
  end: '2024-10-10T01:00:00+03:00'
  value: -0.001
- start: '2024-10-10T01:00:00+03:00'
  end: '2024-10-10T02:00:00+03:00'
  value: -0.073
- start: '2024-10-10T02:00:00+03:00'
  end: '2024-10-10T03:00:00+03:00'
  value: -0.102
- start: '2024-10-10T03:00:00+03:00'
  end: '2024-10-10T04:00:00+03:00'
  value: -0.103
- start: '2024-10-10T04:00:00+03:00'
  end: '2024-10-10T05:00:00+03:00'
  value: -0.1
- start: '2024-10-10T05:00:00+03:00'
  end: '2024-10-10T06:00:00+03:00'
  value: -0.039
- start: '2024-10-10T06:00:00+03:00'
  end: '2024-10-10T07:00:00+03:00'
  value: 0
- start: '2024-10-10T07:00:00+03:00'
  end: '2024-10-10T08:00:00+03:00'
  value: 1.268
- start: '2024-10-10T08:00:00+03:00'
  end: '2024-10-10T09:00:00+03:00'
  value: 2.845
- start: '2024-10-10T09:00:00+03:00'
  end: '2024-10-10T10:00:00+03:00'
  value: 3.028
- start: '2024-10-10T10:00:00+03:00'
  end: '2024-10-10T11:00:00+03:00'
  value: 3.141
- start: '2024-10-10T11:00:00+03:00'
  end: '2024-10-10T12:00:00+03:00'
  value: 3.135
- start: '2024-10-10T12:00:00+03:00'
  end: '2024-10-10T13:00:00+03:00'
  value: 3.105
- start: '2024-10-10T13:00:00+03:00'
  end: '2024-10-10T14:00:00+03:00'
  value: 2.74
- start: '2024-10-10T14:00:00+03:00'
  end: '2024-10-10T15:00:00+03:00'
  value: 2
- start: '2024-10-10T15:00:00+03:00'
  end: '2024-10-10T16:00:00+03:00'
  value: 2.482
- start: '2024-10-10T16:00:00+03:00'
  end: '2024-10-10T17:00:00+03:00'
  value: 1.884
- start: '2024-10-10T17:00:00+03:00'
  end: '2024-10-10T18:00:00+03:00'
  value: 1.751
- start: '2024-10-10T18:00:00+03:00'
  end: '2024-10-10T19:00:00+03:00'
  value: 2.007
- start: '2024-10-10T19:00:00+03:00'
  end: '2024-10-10T20:00:00+03:00'
  value: 1.851
- start: '2024-10-10T20:00:00+03:00'
  end: '2024-10-10T21:00:00+03:00'
  value: 0.25
- start: '2024-10-10T21:00:00+03:00'
  end: '2024-10-10T22:00:00+03:00'
  value: 0
- start: '2024-10-10T22:00:00+03:00'
  end: '2024-10-10T23:00:00+03:00'
  value: -0.001
- start: '2024-10-10T23:00:00+03:00'
  end: '2024-10-11T00:00:00+03:00'
  value: -0.013
- start: '2024-10-10T00:00:00+03:00'
```

and the list goes on, it's very long.

**Comment by Hellowlol** [comment]
Let me know when it ready, and it can create a new release

**Comment by dana-se** [comment]
> Ok I think I figured out what happened. Dateutils parsed timestamps as UTC from API, even though they're Europe/Stockholm. I added an override for parsing dates and now at least FI prices align with what other sources are reporting.

Just merged this, and today's prices are right (SE4). Hoping to see that tomorrow's prices works too, in about an hour.

**Comment by developerfromjokela** [comment]
> > Ok I think I figured out what happened. Dateutils parsed timestamps as UTC from API, even though they're Europe/Stockholm. I added an override for parsing dates and now at least FI prices align with what other sources are reporting.
> 
> 
> 
> Just merged this, and today's prices are right (SE4). Hoping to see that tomorrow's prices works too, in about an hour. 
> 
> 
> 
> Update: Tomorrow's prices just arrived, ad everything looks OK

Same here. I have two instances of nordpool planner. First still has that raw_today issue, second one doesn't. It doesn't really affect functionality otherwise. 

I guess it's good for merging, I can investigate meanwhile what's the issue with my raw_today.

**Comment by dana-se** [comment]
@developerfromjokela don't you think it's time to ask @Hellowlol to make a new release.
Because the old API will probably not work on Monday, 14th October 2024, just to give people a chance to install it before.

Thanks for the excellent work with this code!

**Comment by developerfromjokela** [comment]
> @developerfromjokela don't you think it's time to ask @Hellowlol to make a new release.
> 
> Because the old API will probably not work on Monday, 14th October 2024, just to give people a chance to install it before.
> 
> 
> 
> Thanks for the excellent work with this code!

I agree, core functions work fine now. Minor things can be fixed in another update.

@Hellowlol now can release it

**Review by Teme-V — APPROVED** [review]
Pulled, tested and no found issues

**Comment by Hellowlol** [comment]
@developerfromjokela  and @dana-se Some user reported that one hour is missing. See https://github.com/custom-components/nordpool/issues/464 are we sure the data isnt in UTC? it me it seems like the new API correctly is infact UTC but i would like a second opition. If it is UTC is would simplify things alot

**Comment by developerfromjokela** [comment]
> @developerfromjokela  and @dana-se Some user reported that one hour is missing. See https://github.com/custom-components/nordpool/issues/464 are we sure the data isnt in UTC? it me it seems like the new API correctly is infact UTC but i would like a second opition. If it is UTC is would simplify things alot 

Hmm well from what I checked during rewrite it seemed like CET. Can you save server responses from today or just post them here? I'll check out later.

**Pull request merged**